Tests/UnitTests/Networking/Backend/BackendGetCustomerInfoTests.swift

```diff
@@ -117,15 +117,22 @@ class BackendGetCustomerInfoTests: BaseBackendTests {
             response: .init(statusCode: .success, response: ["sjkaljdklsjadkjs": ""])
         )
 
-        var result: Result<CustomerInfo, NSError>?
+        var result: Result<CustomerInfo, BackendError>?
 
         backend.getCustomerInfo(appUserID: Self.userID) {
-            result = $0.mapError { $0 as NSError }
+            result = $0
         }
 
         expect(result).toEventuallyNot(beNil())
-        expect(result?.error?.domain) == RCPurchasesErrorCodeDomain
-        expect(result?.error?.code) == ErrorCode.unknownBackendError.rawValue
+
+        switch result {
+        case .failure(.networkError(.decoding)):
```

Comment:
Using a `switch` because we can't really copy the underlying error for the decoding error which has a bunch of `JSONDecoder` implementation details. Instead, just checking that it's a `NetworkError.decoding`.

Reply:
Oh yeah much better since we only care about the correct `case`.